src/server/lib/http-request.js

```diff
@@ -0,0 +1,123 @@
+import originalFetch from "node-fetch";
+import AbortController from "node-abort-controller";
+import { log } from "../../lib";
+import { v4 as uuid } from "uuid";
+
+const requestWithRetry = async (url, props) => {
+  const originalProps = props || {};
+  const remainingProps = {
+    ...(originalProps || {})
+  };
+  delete remainingProps.validateStatus;
+  delete remainingProps.retries;
+  delete remainingProps.timeout;
```

Comment:
Everything above this (and a little below) seems like it would be much easier to read with ES6 default deconstructed args, e.g. `const requestWithRetry = async (url, { timeout=2000, retries=2, validateStatus=false }) =>`...etc

Reply:
SHA 3dffe9ec9ec37691d42b25610d4deca98ad15c94